src/components/popover/tab/keymap/keymap-tab.component.ts

```diff
@@ -1,73 +1,71 @@
-import {Component, Input, OnInit } from '@angular/core';
+import { ChangeDetectionStrategy, Component, Input, OnInit } from '@angular/core';
 
-import {Select2OptionData} from 'ng2-select2/ng2-select2';
+import { Select2OptionData } from 'ng2-select2/ng2-select2';
 
-import {KeyAction, SwitchKeymapAction} from '../../../../config-serializer/config-items/key-action';
-import {Keymap} from '../../../../config-serializer/config-items/Keymap';
-import {Tab} from '../tab';
-
-import {UhkConfigurationService} from '../../../../services/uhk-configuration.service';
+import { KeyAction, SwitchKeymapAction } from '../../../../config-serializer/config-items/key-action';
+import { Keymap } from '../../../../config-serializer/config-items/Keymap';
+import { Tab } from '../tab';
 
 @Component({
     selector: 'keymap-tab',
     template: require('./keymap-tab.component.html'),
-    styles: [require('./keymap-tab.component.scss')]
+    styles: [require('./keymap-tab.component.scss')],
+    changeDetection: ChangeDetectionStrategy.OnPush
 })
 export class KeymapTabComponent implements OnInit, Tab {
     @Input() defaultKeyAction: KeyAction;
+    @Input() keymaps: any;
 
-    private keymaps: Keymap[];
     private keymapOptions: Array<Select2OptionData>;
-    private selectedKeymapIndex: number;
+    private selectedKeymap: Keymap;
+    private selectedKeymapIndex: string;
```

Comment:
Yes, but I think that we need it for scenario when you don't have key map selected (state when value is `-1`).